gravitee-apim-console-webui/src/management/api/api-traffic-v4/webhook-logs-details/webhook-logs-details.component.ts

```diff
@@ -150,72 +213,122 @@ export class WebhookLogsDetailsComponent implements OnInit {
+    const reqBody = log.additionalMetrics[REQ_BODY_KEY];
+    const respBody = log.additionalMetrics[RESP_BODY_KEY];
+
+    if (log.status === 0) {
+      this.connectionFailure = {
+        errorKey: log.errorKey ?? null,
+        componentType: log.errorComponentType ?? null,
+        componentName: log.errorComponentName ?? null,
+        message: log.message ?? null,
+        diagnostics: log.warnings ?? [],
+        lastError: typeof lastError === 'string' ? lastError : null,
+      };
+    } else {
+      this.connectionFailure = null;
+    }
+
     this.deliveryAttemptsDataSource = this.buildDeliveryAttempts(log);
-    const attemptsCount = this.deliveryAttemptsDataSource.length || (log.additionalMetrics?.['int_webhook_retry-count'] ?? 1);
+    const attemptsCount = this.deliveryAttemptsDataSource.length || (isNumber(retryCount) ? retryCount : 1);
 
     this.overviewRequest = [
-      { label: 'Date', value: this.formatDate(log.timestamp) },
+      { label: 'Date', value: this.formatDateTime(log.timestamp) },
       { label: 'Method', value: log.method },
       { label: 'Attempts', value: attemptsCount },
       { label: 'Callback URL', value: log.callbackUrl ?? log.uri },
     ];
 
     this.overviewResponse = [
       { label: 'Status', value: log.status },
-      { label: 'Total duration', value: log.duration || `${(log.gatewayResponseTime / 1000).toFixed(1)} s` },
+      {
+        label: 'Total duration',
+        value: log.duration || `${(log.gatewayResponseTime / 1000).toFixed(1)} s`,
+      },
       { label: 'Payload size', value: this.formatPayloadSize(log) },
-      { label: 'Last error', value: log.additionalMetrics?.['string_webhook_last-error'] ?? '—' },
+      { label: 'Last error', value: typeof lastError === 'string' ? lastError : '—' },
     ];
 
-    this.requestHeaders = this.parseHeaders(log.additionalMetrics?.['json_webhook_req-headers']);
-    this.responseHeaders = this.parseHeaders(log.additionalMetrics?.['json_webhook_resp-headers']);
-    this.requestBody = log.additionalMetrics?.['string_webhook_req-body'] ?? '';
-    this.responseBody = log.additionalMetrics?.['string_webhook_resp-body'] ?? '';
-    this.connectionFailure = this.buildConnectionFailure(log);
+    this.requestHeaders = this.parseHeaders(typeof reqHeaders === 'string' ? reqHeaders : null);
+
+    this.responseHeaders = this.parseHeaders(typeof respHeaders === 'string' ? respHeaders : null);
+    this.requestBody = typeof reqBody === 'string' ? reqBody : '';
+
+    this.responseBody = typeof respBody === 'string' ? respBody : '';
+
+    this.cdr.markForCheck();
   }
 
   private buildDeliveryAttempts(log: WebhookLog): DeliveryAttempt[] {
-    const rawTimeline = log.additionalMetrics?.['json_webhook_retry-timeline'];
-    if (!rawTimeline || rawTimeline === '[]') {
+    const rawTimeline = log.additionalMetrics[RETRY_TIMELINE_KEY];
+
+    if (typeof rawTimeline !== 'string' || rawTimeline.trim() === '' || rawTimeline === '[]') {
```

Comment:
It think this change could works:

```suggestion
    if (typeof rawTimeline !== 'string' || !rawTimeline.trim() || rawTimeline === '[]') {
    // rawTimeline === '[]' --> is a string and you already checked if  typeof rawTimeline !== 'string' 🤔 
    
    // Or inverse the if block to make it more readable
     if (typeof rawTimeline === 'string' || rawTimeline.trim()) {
     
     // Maybe have a look at https://www.learn-ts.org/en/Truthy_and_Falsy 
```

Reply:
Done! [fcbd571](https://github.com/gravitee-io/gravitee-api-management/pull/14303/commits/fcbd571882cefbb6a80b68352efc6ac594d43cd6)